### PR TITLE
[Core] Unify argument parsing between the MPI and regular tests

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Unify argument parsing between the MPI and regular tests

**🆕 Changelog**

- [Refactor run_tests with argsparse](https://github.com/KratosMultiphysics/Kratos/commit/9c3582a345287f0e1bacf6dfec2d776267b39687)
- [Use multiprocessing in mpi tests](https://github.com/KratosMultiphysics/Kratos/commit/17e3a57114470831d48e1a53f24dd337297e5c38)
- [Adding help in mpi tests](https://github.com/KratosMultiphysics/Kratos/commit/a22f01b6459dc71842df4388b9357d3fc88259b2)
- [Adding application to mpi tests](https://github.com/KratosMultiphysics/Kratos/commit/b797eb1c072e37c5dbc5c6cc33d04da998a9a29c)

### CODE REVIEW

File: kratos/python_scripts/testing/run_tests.py [hunk 1]
```diff
@@ -181,108 +161,53 @@ def print_summary(exit_codes):
...
-        signalTime = timer
+    if int(args.timer) > 0:
+        signalTime = int(args.timer)
     else:
-        if level == 'small':
-            signalTime = int(90)
-        elif level == 'nightly':
-            signalTime = int(900)
+        if args.level == 'small':
+            signalTime = 90
```
Comment (roigcarlo) [inline kratos/python_scripts/testing/run_tests.py]: I would prefer to keep the cast here if possible- It gives problem with the signals in win (see https://github.com/KratosMultiphysics/Kratos/commit/dbf821e7b468cf85d2150abf92a014c857752f0d)

Comment (loumalouomega) [reply kratos/python_scripts/testing/run_tests.py]: Okay

**Review by loumalouomega — COMMENTED** [review]
Revert cast done

File: kratos/python_scripts/testing/run_tests.py (hunk 1 above)
Comment (loumalouomega) [reply kratos/python_scripts/testing/run_tests.py]: Done

**Comment by loumalouomega** [comment]
> Revert cast done

@roigcarlo por fa plis

**Review by roigcarlo — APPROVED** [review]

**Pull request merged**